pkg/motor/create_account.go

```diff
@@ -11,20 +11,42 @@ import (
 	"github.com/sonr-io/sonr/pkg/did/ssi"
 	"github.com/sonr-io/sonr/pkg/tx"
 	"github.com/sonr-io/sonr/pkg/vault"
+	"github.com/sonr-io/sonr/third_party/types/common"
 	mt "github.com/sonr-io/sonr/third_party/types/motor/api/v1"
 	rt "github.com/sonr-io/sonr/x/registry/types"
 )
 
 func (mtr *motorNodeImpl) CreateAccount(request mt.CreateAccountRequest) (mt.CreateAccountResponse, error) {
 	// create DSC and store it in keychain
+	mtr.triggerWalletCreateEvent(common.WalletEvent{
+		Type:         common.WALLET_EVENT_TYPE_KEY_CREATE_START,
+		ErrorMessage: "",
+		HasErrored:   false,
+		Message:      "",
+		Completed:    false,
+	})
 	dsc, err := kr.CreateDSC()
 	if err != nil {
+		mtr.triggerWalletCreateEvent(common.WalletEvent{
+			Type:         common.WALLET_EVENT_TYPE_KEY_CREATE_END,
+			HasErrored:   true,
+			ErrorMessage: err.Error(),
+			Message:      "",
+			Completed:    true,
```

Comment:
Does this mean the event is completed or the process it's around? It looks like you have an event type specifically denoting the "end" of this process, so the boolean here is redundant.

Reply:
I also think an error can be determined by the presence of `ErrorMessage`, and the `HasErrored` wouldn't be necessary

Reply:
that makes sense, the completion flags are a bit redundant. ill remove them